cd/auto-generate/auto-generate-controllers.sh

```
@@ -90,6 +112,22 @@ else
   echo "auto-generate-controllers.sh][INFO] go version in code-generator/go.mod file is $GO_VERSION_IN_GO_MOD"
 fi
 
+# Get Labels for Release PR
+PR_LABELS=$(get_pr_labels)
+
+# Set type of version bump to apply to downstream controller releases.
+CONTROLLER_RELEASE_TYPE=""
+if echo "$labels" | grep -q "version/patch"; then
+    CONTROLLER_RELEASE_TYPE="patch"
+elif echo "$labels" | grep -q "version/minor"; then
+    CONTROLLER_RELEASE_TYPE="minor"
+else 
+    echo "auto-generate-controllers.sh][ERROR] No labels specifying controller release type found exiting."
```

would failing postsubmit job be too aggressive? maybe we can block merging release PR instead unless there is a version defined

---

or we can use patch by default, unless there is the `minor` label..

---

I considered just patching by default. The reason I leaned towards failing early is that I wanted the developer making the release to actively make choose what kind of downstream version bump should be used. If someone forgets to add the label they can go back and add it after to the PR and restart the job.

---

restarting a job is not really the best experience..maybe we can add a `hold/includeDownstreamVersionBeforeMerge` label when PR is created by default that blocks merge (added here: https://github.com/aws-controllers-k8s/test-infra/blob/f6c882880bbca6c7e248923077b701879157ee44/prow/config/templates/config-ConfigMap.yaml#L97-L101).

---

Maybe, how fine grained is that tooling? We really only want to have that requirement for release PRs. The labels don't have any meaning to most code-gen PRs.